client/components/board.jsx

```diff
@@ -36,10 +38,19 @@ class Board extends React.Component {
     }
     const { row, col } = this.state.selected;
     const input = this.state.challenge;
-    input[row][col] = event.target.value;
+    this.state.previousMove.push({ challenge: JSON.parse(JSON.stringify(input)), selected: JSON.parse(JSON.stringify(this.state.selected)) });
+    input[row][col] = parseInt(event.target.value);
```

Comment:
Ok, there are a few issues here.
- You should not use `JSON.parse(JSON.stringify())` to make copies. it is very inefficient.
- you are pushing into a state array, always use concat() instead

to copy a 2D array
```jsx
const copy = original.map(row => row.map(value => value));
```